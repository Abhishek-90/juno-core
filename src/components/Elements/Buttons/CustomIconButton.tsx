/* eslint-disable react/button-has-type */
import React from 'react'
import styled from 'styled-components'
import * as theme from '../../../constants/themeConstants'

interface ICustomIconButton {
    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
    type?: 'submit' | 'reset' | 'button'
    disabled?: boolean
    icon: {}
    style?: React.CSSProperties
    title?: string,
    className?: string
    isActive?: boolean
    // aria-describedby?: null
}

interface IButton {
    isActive: boolean | undefined
}

const Button = styled.button<IButton>`
  border: none;
  color: ${ (props) => props.isActive ? `${ theme.colorBlack } ` : `${ theme.colorTextLightGrey } ` };
  outline: none;
  background-color: transparent;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out;

  &:hover {
    color: ${ theme.colorBlack };
    cursor: pointer;
  }
`

const CustomIconButton = (props: ICustomIconButton) => {
    const { onClick, disabled, icon, type, style, title, className, isActive } = props
    return (
        <Button
            onClick={(event) => onClick(event)}
            type={type ?? 'button'}
            disabled={disabled}
            style={style}
            title={title}
            className={className}
            isActive={isActive}
        >
            <span>{icon}</span>
        </Button>
    )
}

export default CustomIconButton

CustomIconButton.defaultProps = {
    className: null,
    disabled: false,
    style: null,
    title: null,
    type: 'button',
    isActive: false
}