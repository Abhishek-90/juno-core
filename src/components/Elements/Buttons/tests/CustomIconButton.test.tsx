/* eslint-disable no-console */
import { FiMenu } from 'react-icons/fi'
import renderer from 'react-test-renderer'
import toJson from '../../../../utils/toJSONforTest'
import CustomIconButton from '../CustomIconButton'

test('Button renders with icon if provided', () => {
  const component = renderer.create(
    <CustomIconButton icon={<FiMenu />} onClick={() => {}} title="Test" />
  )
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Button fires the onClick event when clicked', () => {
  const component = renderer.create(
    <CustomIconButton
      icon={<FiMenu />}
      onClick={() => {
        console.log('testing')
      }}
      title="Test"
    />
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onClick()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
