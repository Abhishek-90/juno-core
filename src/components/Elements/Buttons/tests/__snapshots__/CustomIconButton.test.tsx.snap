// Vitest Snapshot v1

exports[`Button fires the onClick event when clicked 1`] = `
<button
  className="sc-bczRLJ ioTdbI"
  disabled={false}
  onClick={[Function]}
  style={null}
  title="Test"
  type="button"
>
  <span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </span>
</button>
`;

exports[`Button fires the onClick event when clicked 2`] = `
<button
  className="sc-bczRLJ ioTdbI"
  disabled={false}
  onClick={[Function]}
  style={null}
  title="Test"
  type="button"
>
  <span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </span>
</button>
`;

exports[`Button renders with icon if provided 1`] = `
<button
  className="sc-bczRLJ ioTdbI"
  disabled={false}
  onClick={[Function]}
  style={null}
  title="Test"
  type="button"
>
  <span>
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </span>
</button>
`;
